### PR TITLE
docs(retry): add stop retrying jobs pattern

### PR DESCRIPTION
**Pull request by roggervalf**
ref https://github.com/taskforcesh/bullmq/issues/1571

### CODE REVIEW

File: docs/gitbook/patterns/stop-retrying-jobs.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+```typescript
+import { Worker, UnrecoverableError } from 'bullmq';
+
+const worker = new Worker(
+  'myQueue',
+  async job => {
+    const [isRateLimited, duration] = await doExternalCall();
+    if (isRateLimited) {
+      await worker.rateLimit(duration);
+      if (job.attemptsMade >= job.opts.attempts) {
```
Comment (manast) [inline docs/gitbook/patterns/stop-retrying-jobs.md:38]: Can't we do this check *before* ```await worker.rateLimit(duration)``` ?

File: tests/test_rate_limiter.ts [hunk 2]
```diff
@@ -377,6 +383,56 @@ describe('Rate Limiter', function () {
...
+    describe('when reaching max attempts and we want to move the job to failed', () => {
+      it('should throw Unrecoverable error', async function () {
+        const dynamicLimit = 550;
+        const duration = 100;
+
+        const worker = new Worker(
+          queueName,
+          async job => {
+            await worker.rateLimit(dynamicLimit);
+            if (job.attemptsMade >= job.opts.attempts!) {
```
Comment (manast) [inline tests/test_rate_limiter.ts:395]: Same comment as above.

File: docs/gitbook/patterns/stop-retrying-jobs.md (hunk 1 above)
Comment (roggervalf) [reply docs/gitbook/patterns/stop-retrying-jobs.md:38]: We can, but it's better to set the queue in rate limit before so other jobs after this one Will not be picked

File: tests/test_rate_limiter.ts (hunk 2 above)
Comment (roggervalf) [reply tests/test_rate_limiter.ts:395]: Same as above

**Review by manast — APPROVED** [review]

**Pull request merged**